### ISSUE TITLE
certain permutations of monomials in two-term approx can result in the same approximations (breaking test_check_if_permutation_exists). 

### ISSUE THREAD

**Issue by 1ozturkbe** [convexengineering/robust#31]
It seems the randomness of this test causes occasional errors.

**Comment by 1ozturkbe** [comment]
Here is an example. First, here is a permutation of monomials (temp), and I know for a fact that it doesn't exist in a set of 100 random permutations. And there is also another permutation, list_of_permutations[23], which is clearly different. 
```python
temp = [3, 2, 0, 1, 6, 5, 8, 7, 4]
list_of_permutations[23] = [2, 3, 0, 1, 6, 5, 7, 8, 4]
```
However, when I compare the two-term posynomials generated by these permutations, they are identical!
```python 
 data_posynomial = 
[gpkit.Posynomial((1.73*x_0^-2.1*x_10^4.2*x_11^2.7*x_12^4.2*x_13^0.98*x_14^-1.1*x_15^-4.9*x_16^-3.5*x_17^-2.5*x_1^1.3*x_2^0.5*x_3^4.8*x_4^-2*x_5^0.14*x_6^4.5*x_7^1.2*x_8^-2.8*x_9^2.4*z^0_1^-1 + 8.16*x_0^0.073*x_10^2.5*x_11^4.7*x_12^1.2*x_13^0.46*x_14^1.6*x_15^4.8*x_16^-4*x_17^1.6*x_1^2.8*x_2^4.1*x_3^-0.54*x_4^-3.5*x_5^0.97*x_6^3*x_7^-0.74*x_8^1.9*x_9^4.8*z^0_1^-1)*z^0_1),
 gpkit.Posynomial((1.11*x_0^-4.5*x_10^3.1*x_11^2.2*x_12^-2.8*x_13^4.9*x_14^4.9*x_15^0.12*x_16^-3.1*x_17^3.5*x_1^1.6*x_2^-4.2*x_3^3.6*x_4^1.7*x_5^4.9*x_6^4.4*x_7^0.71*x_8^2.6*x_9^-0.21*z^1_1^-1 + 8.1*x_0^-3.4*x_10^-1.7*x_11^-1.5*x_12^-3*x_13^-3.1*x_14^0.43*x_15^4.6*x_16^-4.8*x_17^-3.2*x_1^-4*x_2^4.9*x_3^1.4*x_4^3.1*x_5^1.8*x_6^-4.2*x_7^4.2*x_8^0.71*x_9^0.28*z^1_1^-1)*z^1_1),
 gpkit.Posynomial((0.925*x_0^-4.2*x_10^-1.7*x_11^-3.2*x_12^-3.6*x_13^3.7*x_14^3.5*x_15^-3.6*x_16^2.4*x_17^-0.022*x_1^-1.9*x_2^2.4*x_3^2.6*x_4^2.4*x_5^-4.6*x_6^-0.12*x_7^4.3*x_8^-1.2*x_9^-1.3*z^2_1^-1 + 9.62*x_0^1.7*x_10^0.45*x_11^-0.84*x_12^-2.4*x_13^-2.2*x_14^4.3*x_15^2.1*x_16^-2.3*x_17^4.9*x_1^1.1*x_2^2.6*x_3^1.2*x_4^-1.9*x_5^-4.5*x_6^0.94*x_7^-4.6*x_8^4.8*x_9^4*z^2_1^-1)*z^2_1),
 gpkit.Posynomial((3.39*x_0^2.4*x_10^-0.13*x_11^-3.9*x_12^-4.7*x_13^-3*x_14^1.8*x_15^4.6*x_16^1.2*x_17^-2.4*x_1^4.3*x_2^1.7*x_3^-1.6*x_4^-0.65*x_5^4.1*x_6^0.23*x_7^0.89*x_8^-1.3*x_9^4.4*z^3_1^-1 + 7.71*x_0^2.9*x_10^0.34*x_11^-1.9*x_12^-0.67*x_13^3.6*x_14^1*x_15^-2.2*x_16^-1.1*x_17^-1.6*x_1^1.7*x_2^-0.22*x_3^-3.1*x_4^-4.1*x_5^-2.5*x_6^-3.7*x_7^-1.2*x_8^3.2*x_9^-3*z^3_1^-1)*z^3_1),
 gpkit.Monomial(7.15*x_0^-3.2*x_10^4.7*x_11^4*x_12^4.6*x_13^-3.4*x_14^-0.2*x_15^-4.3*x_16^-3.3*x_17^0.54*x_1^-3.3*x_2^-3.3*x_3^3.1*x_4^3*x_5^-0.19*x_6^-2.6*x_7^-3.3*x_8^0.77*x_9^2.1*z^4_1^-1*z^4_1)] %and
 list_of_posynomials[23] = [gpkit.Posynomial((1.73*x_0^-2.1*x_10^4.2*x_11^2.7*x_12^4.2*x_13^0.98*x_14^-1.1*x_15^-4.9*x_16^-3.5*x_17^-2.5*x_1^1.3*x_2^0.5*x_3^4.8*x_4^-2*x_5^0.14*x_6^4.5*x_7^1.2*x_8^-2.8*x_9^2.4*z^0_1^-1 + 8.16*x_0^0.073*x_10^2.5*x_11^4.7*x_12^1.2*x_13^0.46*x_14^1.6*x_15^4.8*x_16^-4*x_17^1.6*x_1^2.8*x_2^4.1*x_3^-0.54*x_4^-3.5*x_5^0.97*x_6^3*x_7^-0.74*x_8^1.9*x_9^4.8*z^0_1^-1)*z^0_1),
 gpkit.Posynomial((1.11*x_0^-4.5*x_10^3.1*x_11^2.2*x_12^-2.8*x_13^4.9*x_14^4.9*x_15^0.12*x_16^-3.1*x_17^3.5*x_1^1.6*x_2^-4.2*x_3^3.6*x_4^1.7*x_5^4.9*x_6^4.4*x_7^0.71*x_8^2.6*x_9^-0.21*z^1_1^-1 + 8.1*x_0^-3.4*x_10^-1.7*x_11^-1.5*x_12^-3*x_13^-3.1*x_14^0.43*x_15^4.6*x_16^-4.8*x_17^-3.2*x_1^-4*x_2^4.9*x_3^1.4*x_4^3.1*x_5^1.8*x_6^-4.2*x_7^4.2*x_8^0.71*x_9^0.28*z^1_1^-1)*z^1_1),
 gpkit.Posynomial((0.925*x_0^-4.2*x_10^-1.7*x_11^-3.2*x_12^-3.6*x_13^3.7*x_14^3.5*x_15^-3.6*x_16^2.4*x_17^-0.022*x_1^-1.9*x_2^2.4*x_3^2.6*x_4^2.4*x_5^-4.6*x_6^-0.12*x_7^4.3*x_8^-1.2*x_9^-1.3*z^2_1^-1 + 9.62*x_0^1.7*x_10^0.45*x_11^-0.84*x_12^-2.4*x_13^-2.2*x_14^4.3*x_15^2.1*x_16^-2.3*x_17^4.9*x_1^1.1*x_2^2.6*x_3^1.2*x_4^-1.9*x_5^-4.5*x_6^0.94*x_7^-4.6*x_8^4.8*x_9^4*z^2_1^-1)*z^2_1),
 gpkit.Posynomial((3.39*x_0^2.4*x_10^-0.13*x_11^-3.9*x_12^-4.7*x_13^-3*x_14^1.8*x_15^4.6*x_16^1.2*x_17^-2.4*x_1^4.3*x_2^1.7*x_3^-1.6*x_4^-0.65*x_5^4.1*x_6^0.23*x_7^0.89*x_8^-1.3*x_9^4.4*z^3_1^-1 + 7.71*x_0^2.9*x_10^0.34*x_11^-1.9*x_12^-0.67*x_13^3.6*x_14^1*x_15^-2.2*x_16^-1.1*x_17^-1.6*x_1^1.7*x_2^-0.22*x_3^-3.1*x_4^-4.1*x_5^-2.5*x_6^-3.7*x_7^-1.2*x_8^3.2*x_9^-3*z^3_1^-1)*z^3_1),
 gpkit.Monomial(7.15*x_0^-3.2*x_10^4.7*x_11^4*x_12^4.6*x_13^-3.4*x_14^-0.2*x_15^-4.3*x_16^-3.3*x_17^0.54*x_1^-3.3*x_2^-3.3*x_3^3.1*x_4^3*x_5^-0.19*x_6^-2.6*x_7^-3.3*x_8^0.77*x_9^2.1*z^4_1^-1*z^4_1)]
```
Not a good test unfortunately...

**Comment by 1ozturkbe** [comment]
I know this is the case because in the assertion, flag_one (which checks the permutation) always gives False, and the posynomial comparison gives True.

**Comment by 1ozturkbe** [comment]
I think I will add a random seed that works to make sure tests always run...

**Comment by 1ozturkbe** [comment]
Kind of interesting observation: This almost always occurs when there is a pairwise change in the permutation, eg. [0,1,2,3] or [0,2,1,3].

**Comment by 1ozturkbe** [comment]
This is resolved as of PR #30
